lib/linux.js

```diff
@@ -123,15 +113,39 @@ function setAvailableApps() {
 	});
 }
 
+function isCinnamon() {
+	const args = ['writable', 'org.cinnamon.desktop.background', 'picture-uri'];
+	// output = childProcess.execFile('gsettings', args, ());
+	return cp.execFile('gsettings', args)
+		.then(out => (out.toString().trim() === 'true'))
+		.catch(err => {
+			err.toString(); // do nothing, this error is expected
+		});
+}
+
 exports.get = function get() {
 	if (!availableApps) {
 		return setAvailableApps().then(get);
 	}
 
 	const app = availableApps.find(app => app.get);
 
-	if (/nitrogen/.test(app.cmd)) {
-		app.cmd = '/bin/sed';
+	if (app.cmd === 'nitrogen') {
+		const homeDir = require('os').homedir();
+		const configFile = path.join(homeDir, '.config/nitrogen/bg-saved.cfg');
+		return fs.readFile(configFile, 'utf8').then(config => {
+			config = config.toString().trim().split('\n');
+			const wallpaper = config.find(line => /file=/.test(line));
+			return wallpaper.replace('file=', '');
+		});
```

Comment:
> I just want the one with file=.

That's already covered by the line splitting and `.find`.

> here can be whitespace leading file

`.trim()` it before checking then. `file=` is too loose as if the filename was `foofile=bar.jpg` it would match. Quite unlikely, but I rather be safe.